### PR TITLE
Support for SequencesExt

### LINKED ISSUE #1486: Community modules: rewire SequencesExt
[SequencesExt.tla](https://github.com/tlaplus/CommunityModules/blob/master/modules/SequencesExt.tla)

### PR DESCRIPTION
**Pull request by konnov**
Closes #1486. This PR contains alternative definitions for `SequencesExt` of the community modules. This module required a lot of work. All comments are derived from the original spec: https://github.com/tlaplus/CommunityModules/blob/master/modules/SequencesExt.tla

This PR also extends `MkSeq`, as was required by more complex use cases in `SequencesExt.tla`. These extensions are intended for the internal use.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (Kukovec) [inline src/tla/__rewire_sequences_ext_in_apalache.tla:229]:
```suggestion
IsPrefix(s, t) ==
  LET __LenS == Len(s) IN
  __LenS  <= Len(t) /\ SubSeq(s, 1, __LenS ) = SubSeq(t, 1, __LenS)
```

File: UNRELEASED.md
Comment (thpani) [inline UNRELEASED.md:20]:
```suggestion
 * Support for the community module `SequencesExt`, see #1539
```

File: src/tla/__apalache_folds.tla
Comment (thpani) [inline src/tla/__apalache_folds.tla:37]: Maybe add an explicit description of `__ApalacheMkSeq` or example that does not go via `FunAsSeq`?

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala
Comment (thpani) [inline tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala:90]: `lcap + rcap` may overflow `lcap + rcap`?

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala
Comment (thpani) [inline tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala:95]: `lcap - rcap` may underflow 0?

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala
Comment (thpani) [inline tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala:100]: Overflow?

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (thpani) [inline src/tla/__rewire_sequences_ext_in_apalache.tla:70]: I don't understand when this would be the case? Are we allocating empty `Seq`s with positive capacity?

**Review by Kukovec — APPROVED** [review]
LGTM, but if you want to be consistent w.r.t. shadowing we should rename all parameters of all operators with the `__` prefix as well.

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (thpani) [inline src/tla/__rewire_sequences_ext_in_apalache.tla:186]:
```suggestion
 * Removes the element at position i shortening the length of s by one.
```
I know you want to keep this in sync with the community modules, but this is really confusing.

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (thpani) [inline src/tla/__rewire_sequences_ext_in_apalache.tla:229]:
```suggestion
  Len(s) <= Len(t) /\ s = SubSeq(t, 1, Len(s))
```

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (thpani) [inline src/tla/__rewire_sequences_ext_in_apalache.tla:360]: Will this work in Apalache, i.e., is `Str` a `Seq`?

**Review by thpani — APPROVED** [review]
I have a few comments/questions, otw LGTM

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (konnov) [reply src/tla/__rewire_sequences_ext_in_apalache.tla:49]: Good question. I expect `LessThan` to be a total order.

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (konnov) [reply src/tla/__rewire_sequences_ext_in_apalache.tla:78]: You can use git blame: https://github.com/tlaplus/CommunityModules/commit/b3029e0ce73d58f09c3a3e82fa3a888acb23d7de

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala
Comment (konnov) [reply tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala:90]: Technically, yes. But we will definitely never reach `MAX_INT`.

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (Kukovec) [reply src/tla/__rewire_sequences_ext_in_apalache.tla:229]: Actually, you could probably even do 
```suggestion
  s = SubSeq(t, 1, Len(s))
```

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (Kukovec) [reply src/tla/__rewire_sequences_ext_in_apalache.tla:360]: No, but AFAIK Igor keeps the original comments

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (konnov) [reply src/tla/__rewire_sequences_ext_in_apalache.tla:70]: Yeah, it's tricky. I did not have `SubSeq` originally and Apalache was giving me a weird counterexample. Since we are increasing capacity at every iteration, it can go beyond the length, if the set contains duplicates. I cannot say that I fully understand this example.

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (konnov) [reply src/tla/__rewire_sequences_ext_in_apalache.tla:229]: Good catch. We should probably propagate it to the original spec in the community modules.

File: src/tla/__rewire_sequences_ext_in_apalache.tla
Comment (konnov) [reply src/tla/__rewire_sequences_ext_in_apalache.tla:360]: nope. Strings in Apalache are indivisible. I have to fix this comment.

**Comment by konnov** [comment]
> You can `git push --force` to trigger a build 😉

I tried it and it did not work

**Pull request merged**
